library/src/main/kotlin/com/connectrpc/http/HTTPClientInterface.kt

```diff
@@ -76,11 +76,14 @@ class Stream(
     fun receiveClose() {
         if (!isReceiveClosed.getAndSet(true)) {
             onReceiveClose()
+            // closing receive side implicitly closes send side, too
+            isSendClosed.set(true)
```

Comment:
We don't want `onSendClose()` to be invoked because that will try to half-close the stream (which should fail since the thing is already done at this point). And I don't think we want to call `sendClose()` first (before doing the receive side) because that is just an extra, unnecessary network packet to send the half-close (technically an empty "end-of-stream" data frame) when the cancellation is all that is needed. As the comment says, closing the receive side of the stream implicitly closes the send -- so book-keeping is all that is needed here and no other action. I can add comments to this effect.

Unclear about using try/finally. On the one hand, if `onReceiveClose` fails, then the stream may not be closed so that's an argument to not use finally. However, it doesn't really make much sense to leave `isSendClosed` set to false when `isReceiveClosed` is set to true. So I guess I've talked myself into needing a finally here.